lib/routers/profile/person.js

```diff
@@ -49,6 +49,65 @@ const validateEmail = () => {
   };
 };
 
+const validatePassword = () => {
+  const MIN_CHARS = 10;
+  const MIN_UPPER = 1;
+  const MIN_LOWER = 1;
+  const MIN_DIGIT = 1;
+
+  return (req, res, next) => {
+    const password = get(req, 'body.data.password');
+
+    if (!password) {
+      throw new BadRequestError('A password must be provided');
+    }
+
+    // todo: check password strength
+    if (password.length < MIN_CHARS) {
+      throw new BadRequestError(`Password must be at least ${MIN_CHARS} characters`);
+    }
+
+    if ((password.match(/[A-Z]/g) || []).length < MIN_UPPER) {
+      throw new BadRequestError(`Password must have at least ${MIN_UPPER} uppercase characters`);
+    }
+
+    if ((password.match(/[a-z]/g) || []).length < MIN_LOWER) {
+      throw new BadRequestError(`Password must have at least ${MIN_LOWER} lowercase characters`);
+    }
+
+    if ((password.match(/\d/g) || []).length < MIN_DIGIT) {
+      throw new BadRequestError(`Password must have at least ${MIN_DIGIT} digits`);
+    }
+
```

Comment:
FYI we're duplicating this logic in the UI validation but I thought it best to include it in the API as well...
https://github.com/UKHomeOffice/asl-pages/pull/864/files#diff-b973646600d3a67249d363fef3f9886c

Reply:
100% agree.

Reply:
Although, having said that, it would be interesting to see what keycloak returns for an invalid password. We could be better off having fewer places to update should we ever have cause to change the rules.